express-crud/server.ts

```diff
@@ -0,0 +1,108 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index';
+const app = express();
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+app.use(express.json());
+type Grade = {
+  gradeId?: number;
+  name: string;
+  course: string;
+  score: number;
+};
+function validateGradeId(gradeId: string): void {
+  if (!Number.isInteger(+gradeId)) {
+    throw new ClientError(400, 'gradeId must be interger.');
+  }
+}
+function validateGrade({ name, course, score }: Grade): void {
+  if (!name) throw new ClientError(400, 'name is required');
+  if (!course) throw new ClientError(400, 'course is required');
+  if (!score || !(score >= 0 && score <= 100) || !Number.isInteger(+score))
```

Comment:
It would make sense to check if it's a number _before_ checking if it is >= 0 or <= 100.